src/query.c

```diff
@@ -895,7 +895,11 @@ static IndexIterator *Query_EvalGeometryNode(QueryEvalCtx *q, QueryNode *node) {
   RS_LOG_ASSERT(node->type == QN_GEOMETRY, "query node type should be geometry");
 
   const FieldSpec *fs = node->gmn.geomq->fs;
-  const GeometryIndex *index = OpenGeometryIndex(q->sctx->spec, fs);
+
+  // TODO: open with DONT_CREATE_INDEX once the query string is validated before we get here.
+  // Currently, if  we use DONT_CREATE_INDEX, and the index was not initalized yet, and the query is invalid,
+  // we return results as if the index was empty, instead of raising an error.
```

Comment:
What should be done here? Parse the query before getting here? Fix something in our parser? Can we do something in this PR?

Reply:
Yes. not in the scope of this PR
I'm opening a ticket